File: library/src/main/java/de/timfreiheit/mozart/model/image/CoverImage.java
package de.timfreiheit.mozart.model.image;

import android.graphics.Bitmap;
import android.support.annotation.Nullable;

import com.google.auto.value.AutoValue;

@AutoValue
public abstract class CoverImage {

    @Nullable
    public abstract Bitmap largeImage();

    @Nullable
    public abstract Bitmap icon();

    public static CoverImage create(Bitmap largeImage, Bitmap icon) {
        return new AutoValue_CoverImage(largeImage, icon);
    }

    @SuppressWarnings("ConstantConditions")
    public int byteCount() {
        int size = 0;
        if (largeImage() != null) {
            size += largeImage().getByteCount();
        }
        if (icon() != null) {
            size += icon().getByteCount();
        }
        return size;
    }

}

File: mozart/src/main/java/de/timfreiheit/mozart/model/image/CoverImage.kt
package de.timfreiheit.mozart.model.image

import android.graphics.Bitmap

import com.google.auto.value.AutoValue

data class CoverImage(
    val largeImage: Bitmap?,
    val icon: Bitmap?
) {

    val byteCount: Int by lazy {
        var size = 0
        if (largeImage != null) {
            size += largeImage.byteCount
        }
        if (icon != null) {
            size += icon.byteCount
        }
        size
    }

}
